### DISCUSSION TITLE
[SOLVED] Adding in Quasar with Vite

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [posva/unplugin-vue-router#75] [Questions and Help] [answered]
I'm struggling here to get it working on a Quasar CLI with Vite project.

In the quasar config, in the vitePlugins array, I added  `require('unplugin-vue-router/vite')`.

Then, in the src/router/index.ts, replaced the exported function with the `createRouter` from `vue-router/auto`,
but it complains it doesn't find that module.

I tried reloading VSCode, restarting TypeScript server, deleting node_modules and reinstalling deps with pnpm.
Tried adding the --shamefully-hoist flag to `pnpm install`. Tried ditching pnpm and using npm.

Searched the Issues, the Discussions, Google, nothing.

Any kind answer will be read and appreciated.

**Comment by angelhdzmultimedia** [comment]
### Update

Got it working on a pure Vue 3 + Vite project (with `create-vue` using `pnpm create vue`).

This is the error I get in the browser on the Quasar + Vite project:

![image](https://user-images.githubusercontent.com/10191085/194688889-0da41d55-9cc9-46c5-8ad7-9566850d6761.png)

And in the terminal:

![image](https://user-images.githubusercontent.com/10191085/194688903-c31688da-cc0b-4105-a681-5fdf99ebb88f.png)

**Comment by posva** [comment]
You need to start running the dev server so the d.ts fille is generated

**Comment by angelhdzmultimedia** [comment, marked as answer]
### Solved

`typed-router.d.ts` was there. That's why I was so confused because I was following the guide step by step, even the TypeScript section.

The issue was vitePlugins array in Quasar configuration. It didn't accept the package import as `vitePlugins: [ [ require("unplugin-vue-router/vite") ] ]`. I changed it to `vitePlugins: [ [ "unplugin-vue-router/vite" ] ]` and it worked right away.
